server/src/test/java/io/crate/execution/engine/join/LeftOuterHashJoinBatchIteratorTest.java

```diff
@@ -0,0 +1,165 @@
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.engine.join;
+
+import static com.carrotsearch.randomizedtesting.RandomizedTest.$;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.ToIntFunction;
+
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.carrotsearch.randomizedtesting.RandomizedRunner;
+import com.carrotsearch.randomizedtesting.annotations.Name;
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope;
+
+import io.crate.data.BatchIterator;
+import io.crate.data.Row;
+import io.crate.data.breaker.RowAccounting;
+import io.crate.data.join.CombinedRow;
+import io.crate.data.testing.BatchIteratorTester;
+import io.crate.data.testing.BatchIteratorTester.ResultOrder;
+import io.crate.data.testing.BatchSimulatingIterator;
+import io.crate.data.testing.TestingBatchIterators;
+
+@RunWith(RandomizedRunner.class)
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+public class LeftOuterHashJoinBatchIteratorTest {
+
+    private final CircuitBreaker circuitBreaker = mock(CircuitBreaker.class);
+    private final List<Object[]> expectedResult;
+    private final Supplier<BatchIterator<Row>> leftIterator;
+    private final Supplier<BatchIterator<Row>> rightIterator;
+
+    private static Predicate<Row> getCol0EqCol1JoinCondition() {
+        return row -> Objects.equals(row.get(0), row.get(1));
+    }
+
+    private static ToIntFunction<Row> getHashForLeft() {
+        return row -> Objects.hash(row.get(0));
+    }
+
+    private static ToIntFunction<Row> getHashForRight() {
+        return row -> Objects.hash(row.get(0));
+    }
+
+    private static ToIntFunction<Row> getHashWithCollisions() {
```

Comment:
Can we please add a test for hash collisions as well?

Reply:
Good catch! This was not supported.